core/components/use-templates.tsx

```diff
@@ -0,0 +1,102 @@
+import { useEffect, useState, useCallback } from "react";
```

Comment:
Reducer is ok, but I think this shall go to the `templates` component.

This was actually not require (Is better user experience) but was not part of the bugs.